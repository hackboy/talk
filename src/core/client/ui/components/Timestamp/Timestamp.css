.root {
  composes: button from "talk-ui/shared/typography.css";
  background-color: transparent;
}
