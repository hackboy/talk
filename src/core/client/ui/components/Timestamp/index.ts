export * from "./Timestamp";
export { default } from "./Timestamp";
