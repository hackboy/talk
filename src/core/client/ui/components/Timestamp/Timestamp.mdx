---
name: Timestamp
menu: UI Kit
---

import { Playground } from 'docz'
import Timestamp from './Timestamp'

# Timestamp

## Basic usage
<Playground>
  <Timestamp date={`2018-07-04T12:14:14.564Z`} />
</Playground>

