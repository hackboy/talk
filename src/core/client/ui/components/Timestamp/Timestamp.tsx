import React from "react";
import TimeAgo, { Formatter } from "react-timeago";

import { UIContext } from "talk-ui/components";
import { withStyles } from "talk-ui/hocs";
import { PropTypesOf } from "talk-ui/types";

import * as styles from "./Timestamp.css";

interface InnerProps {
  date: string;
  live?: boolean;
  classes: typeof styles;
  formatter?: Formatter;
}

class Timestamp extends React.Component<InnerProps> {
  public render() {
    const { date, classes, live } = this.props;
    return (
      <UIContext.Consumer>
        {({ timeagoFormatter }) => (
          <TimeAgo
            date={date}
            className={classes.root}
            live={live}
            formatter={timeagoFormatter}
          />
        )}
      </UIContext.Consumer>
    );
  }
}

const enhanced = withStyles(styles)(Timestamp);
export type TimestampProps = PropTypesOf<typeof enhanced>;
export default enhanced;
