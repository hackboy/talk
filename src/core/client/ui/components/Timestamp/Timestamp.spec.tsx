import React from "react";
import { create } from "react-test-renderer";

import Timestamp from "./Timestamp";

it("renders old date", () => {
  const props = {
    date: new Date("December 17, 1995 03:24:00").toISOString(),
  };
  const tree = create(<Timestamp {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders future date", () => {
  const props = {
    date: new Date("December 17, 2108 03:24:00").toISOString(),
  };
  const tree = create(<Timestamp {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
