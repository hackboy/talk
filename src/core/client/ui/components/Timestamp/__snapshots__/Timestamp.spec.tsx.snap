// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders future date 1`] = `
<time
  className="root"
  dateTime="2108-12-17T06:24:00.000Z"
  title="2108-12-17T06:24:00.000Z"
>
  91 years from now
</time>
`;

exports[`renders old date 1`] = `
<time
  className="root"
  dateTime="1995-12-17T06:24:00.000Z"
  title="1995-12-17T06:24:00.000Z"
>
  23 years ago
</time>
`;
